### ISSUE TITLE
better motivation for general inner products

### ISSUE THREAD

**Issue by rumata3** [mml-book/mml-book.github.io#529]
The non-standard inner product in Example 3.5 seems contrived; it's unclear when one would use it.  It would help one's understanding if the usefulness of general inner products beyond the dot product could be better motivated.

Thanks a lot for writing this book and making it available; it's a great aid in re-learning the subjects.

**Comment by mpd37** [comment]
Agree. I'll mark this as a feature request because this goes beyond cosmetic changes that we can make at this stage.
